src/util/transition/ngbTransition.spec.ts

```diff
@@ -2,14 +2,16 @@ import {ngbRunTransition} from './ngbTransition';
 import createSpy = jasmine.createSpy;
 import {Component, ElementRef, ViewChild} from '@angular/core';
 import {ComponentFixture, TestBed} from '@angular/core/testing';
-import {isBrowserVisible} from '../../test/common';
+import {isBrowser, isBrowserVisible} from '../../test/common';
 
 /**
  * This is sometimes necessary only for IE when it fails to recalculate styles synchronously
  * after the 'transitionend' event was fired. To remove when not supporting IE anymore.
  */
-function getComputedStyleAsync(element: HTMLElement, style: keyof CSSStyleDeclaration) {
-  return new Promise<string>(resolve => setTimeout(() => resolve(window.getComputedStyle(element)[style])));
+function getComputedStyleAsync(element: HTMLElement, style: keyof CSSStyleDeclaration): Promise<string> {
+  const getStyle = () => window.getComputedStyle(element)[style];
+  return isBrowser('ie') ? new Promise<string>(resolve => setTimeout(() => resolve(getStyle()), 16)) :
+                           Promise.resolve(getStyle());
```

Comment:
Just another idea on how to manage it here. Instead of having a dedicated function to manage the 'get the computed style later for ie', I would have preferred to introduce a new generic utility, like `wait`. 

I think it would separate the the fact that the browser needs to wait, and the logic that is required after the wait. By the wait, this generic function can be useful if needed in other situation.

In this test file, it would be used like this:

```
import {isBrowser, isBrowserVisible, wait} from '../../test/common';
...
const DELAY = isBrowser('ie') ? 16 : 0;
...
expect(errorSpy).not.toHaveBeenCalled();
await wait(DELAY);
expect(window.getComputedStyle(element).opacity).toBe('0');
```
Instead of 
```
expect(await getComputedStyleAsync(element, 'opacity')).toBe('0');
```

What do you think ?

Reply:
I like your idea much more than mine. However I'll add another `refactor` commit for this instead of fixing all individual commits in this PR